docs/EmbeddedSwift/UserManual.md

```diff
@@ -0,0 +1,138 @@
+# Embedded Swift -- User Manual
+
+**⚠️ Embedded Swift is experimental. This document might be out of date with latest development.**
+
+**‼️ Use the latest downloadable 'Trunk Development' snapshot from swift.org to use Embedded Swift. Public releases of Swift do not yet support Embedded Swift.**
+
+For an introduction and motivation into Embedded Swift, please see "[A Vision for Embedded Swift](https://github.com/apple/swift-evolution/blob/main/visions/embedded-swift.md)", a Swift Evolution document highlighting the main goals and approaches.
+
+The following document explains how to use Embedded Swift's support in the Swift compiler and toolchain.
+
+## What Embedded Swift is, and what it isn't
+
+- Embedded Swift **is** a way to produce small and freestanding binaries (with no, or trivial dependencies).
+- Embedded Swift **is not** a complete one-click solution to program all embedded boards and MCUs.
+- Embedded Swift **is** a compilation model that's analogous to a traditional C compiler in the sense that the compiler produces an object file (.o) that can be simply linked with your existing code, and it's not going to require you to port any libraries or runtimes.
+- Embedded Swift **is not** a HAL, it's not an SDK for development, it's not a set of libraries to program peripherals using high-level APIs. It's instead a compilation mode that's suitable for creating these components.
+
+## Using Embedded Swift
+
+A typical setup and build + run cycle for an embedded development board involves:
+
+- (1) Getting an SDK with the C compilers, headers and libraries for the target
+- (2) Building the C source code, and Swift source code
+- (3) Linking all the libraries, C object files, and Swift object files.
+- (4) Post-processing the linked firmware into a flashable format (UD2, BIN, or bespoke formats)
+- (5) Uploading the flashable binary to the board over a USB cable using some vendor-provided JTAG/SWD tool or by copying it to a fake USB Mass Storage volume presented by the board.
+- (6) Restarting the board, observing physical effects of the firmware (LEDs light up) or UART output over USB, or presence on network, etc.
+
+Most of these steps are out of scope for this document, instead refer to the vendor provided documentation and get familiar with the details of firmware development for your board without Swift in the mix first. Even if you want to build a completely pure Swift firmware, you are still very likely going to need the the vendor provided tooling for linking, post-processing, uploading, etc.
+
+## Building code using Embedded Swift
+
+A basic way to build a set of Swift source files in Embedded Swift mode, is to simply give the compiler (1) a target triple, (2) the `-enable-experimental-feature Embedded` flag, (3) the set of source files that form the input module:
+
+```bash
+$ swiftc -target <target triple> -enable-experimental-feature Embedded \
```

Comment:
Will `-wmo` become optional or implied at some point? Otherwise if this stays as a required flag for embedded, it should be mentioned here.

Reply:
I'll add it here. I think we *should* change the driver to turn WMO on for you, but let's not wait for that to happen.